table/tables/tables.go

```diff
@@ -669,6 +669,7 @@ func (t *TableCommon) addIndices(sctx sessionctx.Context, recordID int64, r []ty
 		}
 		if dupHandle, err := v.Create(sctx, rm, indexVals, recordID, opts...); err != nil {
 			if kv.ErrKeyExists.Equal(err) {
+				txn.CleanupKeyExistErrInfo([]kv.Key{t.RecordKey(recordID)})
```

Comment:
Could we put the clean operations at one place, not quite sure.

Reply:
This line will introduce a new bug in optimistic txn. Here is an example
```sql
create table t (pk int primary key, uk int unique);
insert into t values (1, 2);

begin optimistic;
insert into t values (1, 4);
update t set pk = 2 where uk = 4; -- Duplicate entry '4' for key 'uk'
insert into t values (1, 4);
insert into t values (1, 5);
commit; -- no error

admin check table t; -- ERROR 1105 (HY000): col uk, handle 1, index:types.Datum{k:0x1, collation:"", decimal:0x0, length:0x0, i:2, b:[]uint8(nil), x:interface {}(nil)} != record:types.Datum{k:0x1, collation:"", decimal:0x0, length:0x0, i:5, b:[]uint8(nil), x:interface {}(nil)}
```